grammars/liquid.tmLanguage.json

```
@@ -120,13 +120,16 @@
         {
           "include": "#liquid_doc_example_tag"
         },
+        {
+          "include": "#liquid_doc_description_tag"
+        },
         {
           "include": "#liquid_doc_fallback_tag"
         }
       ]
     },
     "liquid_doc_param_tag": {
-      "match": "(@param)\\s+(?:({[^}]*}?)\\s+)?([a-zA-Z_]\\w*)?",
+      "match": "(@param)\\s+(?:({[^}]*}?)\\s+)?([a-zA-Z_][\\w-]*)?(?:\\s+(.*))?",
```

```suggestion
      "match": "(@param)\\s+(?:({[^}]*}?)\\s+)?(\\[?[a-zA-Z_][\\w-]*\\]?)?(?:\\s+(.*))?",
```

Let's also capture optional params